obelisk/cpp/obelisk_cpp/include/obelisk_mujoco_sim_robot.h

```diff
@@ -0,0 +1,635 @@
+                    }
+
+                    std::string setting = group.substr(0, setting_idx);
+                    std::string val     = group.substr(setting_idx + 1, val_idx - (setting_idx + 1));
+                    setting_map.emplace(setting, val);
+                    group.erase(0, val_idx + val_delim.length());
+                }
+
+                // The dt for the timer
+                double dt = -1;
+                try {
+                    dt = std::stod(setting_map.at("dt"));
+                    if (dt <= 0) {
+                        throw std::runtime_error("Invalid dt. Must be > 0.");
+                    }
+                } catch (const std::exception& e) {
+                    throw std::runtime_error("No dt provided for a sensor!");
+                }
+
+                // The topic for the publisher
+                std::string topic;
+                try {
+                    topic = setting_map.at("topic");
+                } catch (const std::exception& e) {
+                    throw std::runtime_error("No topic provided for a sensor!");
+                }
+
+                const int depth = this->ObeliskNode::GetHistoryDepth(setting_map);
+
+                std::string sensor_type;
+                try {
+                    sensor_type = setting_map.at("sensor_type");
+                } catch (const std::exception& e) {
+                    throw std::runtime_error("No sensor type provided for a sensor!");
+                }
+
+                std::vector<std::string> sensor_names;
+                const std::string name_delim = "&";
+                try {
+                    std::string names = setting_map.at("sensor_names");
+                    while (!names.empty()) {
+                        size_t plus_idx = names.find(name_delim);
+                        if (plus_idx == std::string::npos) {
+                            plus_idx = names.length();
+                        }
+                        sensor_names.emplace_back(names.substr(0, plus_idx));
+                        names.erase(0, plus_idx + name_delim.length());
+                    }
+                } catch (const std::exception& e) {
+                    throw std::runtime_error("No sensor name was provided for a sensor group!");
+                }
+
+                // Create the timer and publishers with the settings
+                if (sensor_type == "jointpos") {
+                    // Make a key
+                    const std::string sensor_key = "sensor_group_" + std::to_string(num_sensors_);
+
+                    // Make a publisher and add it to the list
+                    auto pub = ObeliskNode::create_publisher<obelisk_sensor_msgs::msg::JointEncoders>(topic, depth);
+                    this->publishers_[sensor_key] =
+                        std::make_shared<internal::ObeliskPublisher<obelisk_sensor_msgs::msg::JointEncoders>>(pub);
+
+                    // TODO: Do we want them all in their own reentrant cbg?
```

Comment:
no, they can all belong to the one reentrant cbg. if they're in different cbgs, it's functionally the same thing (regardless of whether the other cbgs are reentrant or mutually exclusive)